prover/src/zk_circuits_handler/bernoulli.rs

```diff
@@ -70,34 +69,50 @@ impl BaseCircuitsHandler {
                 self.get_output_dir(),
             )?;
 
-            return serde_json::to_string(&chunk_proof).map_err(|e| anyhow::anyhow!(e));
+            return Ok(chunk_proof);
         }
         unreachable!("please check errors in proof_type logic")
     }
 
-    fn gen_batch_proof(&self, task: &crate::types::Task) -> Result<String> {
-        let chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)> =
-            self.gen_chunk_hashes_proofs(task)?;
-        let chunk_proofs: Vec<ChunkProof> =
-            chunk_hashes_proofs.iter().map(|t| t.1.clone()).collect();
+    fn gen_chunk_proof(&self, task: &crate::types::Task) -> Result<String> {
+        let chunk_trace = self.gen_chunk_traces(task)?;
+        let chunk_proof = self.gen_chunk_proof_raw(chunk_trace)?;
+        serde_json::to_string(&chunk_proof).map_err(|e| anyhow::anyhow!(e))
+    }
 
+    fn gen_batch_proof_raw(
+        &self,
+        chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)>,
+    ) -> Result<BatchProof> {
         if let Some(prover) = self.batch_prover.as_ref() {
+            let chunk_proofs: Vec<ChunkProof> =
+                chunk_hashes_proofs.iter().map(|t| t.1.clone()).collect();
+
             let is_valid = prover.borrow_mut().check_chunk_proofs(&chunk_proofs);
 
             if !is_valid {
-                bail!("non-match chunk protocol, task-id: {}", &task.id)
+                bail!("non-match chunk protocol")
             }
+
             let batch_proof = prover.borrow_mut().gen_agg_evm_proof(
                 chunk_hashes_proofs,
                 None,
                 self.get_output_dir(),
             )?;
 
-            return serde_json::to_string(&batch_proof).map_err(|e| anyhow::anyhow!(e));
+            return Ok(batch_proof);
         }
         unreachable!("please check errors in proof_type logic")
     }
 
+    fn gen_batch_proof(&self, task: &crate::types::Task) -> Result<String> {
+        log::info!("[circuit] gen_batch_proof for task {}", task.id);
+        let chunk_hashes_proofs: Vec<(ChunkHash, ChunkProof)> =
+            self.gen_chunk_hashes_proofs(task)?;
+        let batch_proof = self.gen_batch_proof_raw(chunk_hashes_proofs)?;
+        serde_json::to_string(&batch_proof).map_err(|e| anyhow::anyhow!(e))
```

Comment:
will `Ok(serde_json::to_string(&batch_proof)?)` work?

Reply:
Yes, it works. Learned a new tip

Reply:
have submitted a new commit fixing this